sdk/storage/Azure.Storage.Blobs/src/Models/BlockListTypes.cs

```diff
@@ -41,14 +41,20 @@ internal static partial class BlobExtensions
         /// </summary>
         /// <param name="options"></param>
         /// <returns>The BlockList response.</returns>
-        internal static BlockListType ToBlockListType(this BlockListOptions options)
+        internal static BlockListType ToBlockListType(this BlockListTypes options)
         {
-            if (options.HasFlag(BlockListOptions.All) || options == default)
-            { return BlockListType.All;  }
-            else if (options.HasFlag(BlockListOptions.Committed))
-            { return BlockListType.Committed; }
+            if (options >= BlockListTypes.All || options == default)
```

Comment:
Nit - This still feels more brittle than catching defaults with the default case.  Here's an example of what I'm talking about:
```
    internal static BlockListType ToBlockListType(this BlockListTypes options) =>
        options switch
        {
            BlockListTypes.Committed   => BlockListType.Committed,
            BlockListTypes.Uncommitted => BlockListType.Uncommitted,
            _                          => BlockListType.All,
        };
```

Reply:
thanks for the "example" :)